contracts/class-type/src/entry.rs

```diff
@@ -1,6 +1,131 @@
+use alloc::vec::Vec;
+use ckb_std::{
+    ckb_constants::Source,
+    ckb_types::{bytes::Bytes, prelude::*},
+    high_level::{load_cell_data, load_script},
+};
 use core::result::Result;
-use script_utils::error::Error;
+use script_utils::{
+    class::{Class, CLASS_TYPE_ARGS_LEN},
+    error::Error,
+    helper::{
+        count_cells_by_type_args, load_cell_data_by_type_args, load_output_type_args_ids, Action,
+    },
+    issuer::{Issuer, ISSUER_TYPE_ARGS_LEN},
+};
 
-pub fn main() -> Result<(), Error> {
+fn check_issuer_args<'a>(class_args: &'a Bytes) -> impl Fn(&Bytes) -> bool + 'a {
+    move |type_args: &Bytes| type_args[..] == class_args[0..ISSUER_TYPE_ARGS_LEN]
+}
+
+fn check_class_args<'a>(class_args: &'a Bytes) -> impl Fn(&Bytes) -> bool + 'a {
+    move |type_args: &Bytes| {
+        type_args.len() == CLASS_TYPE_ARGS_LEN
+            && type_args[0..ISSUER_TYPE_ARGS_LEN] == class_args[0..ISSUER_TYPE_ARGS_LEN]
+    }
+}
+
+fn parse_class_action(class_args: &Bytes) -> Result<Action, Error> {
+    let inputs_count = count_cells_by_type_args(Source::Input, &check_class_args(class_args));
+    let outputs_count = count_cells_by_type_args(Source::Output, &check_class_args(class_args));
+
+    match (inputs_count, outputs_count) {
+        (0, _) => Ok(Action::Create),
+        (1, 1) => Ok(Action::Update),
+        (1, 0) => Ok(Action::Destroy),
+        _ => Err(Error::ClassCellsCountError),
+    }
+}
+
+fn handle_creation(class_args: &Bytes) -> Result<(), Error> {
+    let class_cell_data = load_cell_data(0, Source::GroupOutput)?;
+    let class = Class::from_data(&class_cell_data[..])?;
+    if class.issued != 0 {
+        return Err(Error::ClassIssuedInvalid);
+    }
+
+    let issuer_inputs_count =
+        count_cells_by_type_args(Source::Input, &check_issuer_args(class_args));
+    let issuer_outputs_count =
+        count_cells_by_type_args(Source::Output, &check_issuer_args(class_args));
+    if issuer_inputs_count != 1 || issuer_outputs_count != 1 {
+        return Err(Error::IssuerCellsCountError);
+    }
+
+    let input_issuer =
+        match load_cell_data_by_type_args(Source::Input, &check_issuer_args(class_args)) {
+            Some(data) => Ok(Issuer::from_data(&data[..])?),
+            None => Err(Error::IssuerDataInvalid),
+        }?;
+    let output_issuer =
+        match load_cell_data_by_type_args(Source::Output, &check_issuer_args(class_args)) {
+            Some(data) => Ok(Issuer::from_data(&data[..])?),
+            None => Err(Error::IssuerDataInvalid),
+        }?;
+
+    if output_issuer.class_count < input_issuer.class_count {
```

Comment:
```suggestion
    if output_issuer.class_count <= input_issuer.class_count {
```

Reply:
No `=`. The `class_count` of input can be equal to the `class_count` of output.